lib/pbench/cli/server/users_list.py

```diff
@@ -0,0 +1,68 @@
+import click
+from pbench import BadConfig
+from pbench.server import PbenchServerConfig
+from pbench.cli.server import pass_cli_context
+from pbench.cli.server.options import common_options
+from pbench.common.logger import get_pbench_logger
+from pbench.server.database.models.users import User
+from pbench.server.database.database import Database
+
+_NAME_ = "pbench-user-list"
+
+
+class UsersList:
+    def __init__(self, context):
+        self.context = context
+
+    def execute(self):
+        config = PbenchServerConfig(self.context.config)
+
+        logger = get_pbench_logger(_NAME_, config)
+
+        # We're going to need the Postgres DB to track dataset state, so setup
+        # DB access.
+        Database.init_db(config, logger)
+
+        click.echo(f"First_Name Last_Name Registered_On Email")
+        if not self.context.reversed:
+            users = Database.db_session.query(User).limit(self.context.N).all()
+        else:
+            users = (
+                Database.db_session.query(User)
+                .order_by(User.id.desc())
+                .limit(self.context.N)
+                .all()
+            )
+
+        for user in users:
+            click.echo(
+                f"{user.first_name} {user.last_name} {user.registered_on} {user.email}"
+            )
+
+
+@click.command()
+@common_options
+@click.option(
+    "--N",
+    required=False,
+    default=100,
+    help="Number of user table rows to print, defaults to 100",
+)
+@click.option(
+    "--reversed",
+    type=bool,
+    required=False,
+    default=False,
+    help="Last n rows of user table to print",
```

Comment:
This isn't very useful help!

Reply:
These two options are nice, but I wouldn't bother:  this is a CLI tool which we expect to run under a Linux shell -- the output of this command can be piped to `head`, `tail`, `sort`, `less`, `grep`, et al. -- there's little need to reinvent those wheels.

The only reason I see for functionality like this would be if the output of this tool were likely to be awkwardly large (like larger than we would want to pipe through shell commands), and I think that would be a great kind of problem to have, and we can solve it if it ever arises.

Reply:
@webbnh, `head, tail, sort, less, grep` will sort the already obtained query result, it wont affect the actual query result. What `--reversed` option does is that it queries the database in a reversed fashion giving the last n rows instead of first n rows

Reply:
I'm failing to see why anyone would care about the first or last set of rows where the ordering is based on the database ID.  What am I missing?

`pbench-user-list | sort -r | head` would yield the last ten rows.  (Of course, the "lastness" would be based on first name or something instead of database ID, but we could sort them based on any of the output columns instead....)

Reply:
What I meant was if we do `pbench-user-list -n 20` will give you the first 20 rows from the users table where the user ids are [0, 1, 2 ... 20]. However if we want to list the last 20 rows from the users table [81, 82, 83, ... 100] (If our database has 100 rows) then we could do `pbench-user-list -n 20 --reversed True`

Reply:
I understand.  But, why would we want to do that?

The database IDs are opaque identifiers.  What constitutes "first" is arbitrary.  It's probably creation order, but...if we really care about creation order then we should sort by creation date or we should explicitly specify an ordinal field.

But I don't think we care what order the records appear in the table...do we?  So, why should we have a `--reversed` option?  And, since we have other means for limiting the output, why should we have a `-n` option?

Options come at a cost.  They have to be documented.  They have to be tested.  They require code which has to be maintained.  They tend to accumulate a compatibility burden.

It's nice to know that SQLAlchemy offers these capabilities.  But, it doesn't mean that we _have_ to make use of them.  Just be cause we _can_ do something doesn't mean that we _should_.  If we don't have a requirement for a feature, it's probably best to wait to provide it, especially if it's not hard to add it later.

Reply:
Understood. I agree `-n` option is not really helpful when we already have linux commands to limit the results. And therefore if we aren't using the `-n` option `--reversed` option becomes useless.